core/src/androidMain/kotlin/Scanner.kt

```diff
@@ -53,22 +56,34 @@ public class AndroidScanner internal constructor(
             }
 
             override fun onScanFailed(errorCode: Int) {
+                logger.error { message = "Scan could not be started, error code $errorCode." }
                 cancel("Bluetooth scan failed", ScanFailedException(errorCode))
             }
         }
 
-        val scanFilter =
-            filterServices
-                ?.map { ScanFilter.Builder().setServiceUuid(ParcelUuid(it)).build() }
-                ?.toList()
-        bluetoothAdapter.bluetoothLeScanner.startScan(
-            scanFilter,
-            scanSettings,
-            callback,
-        )
+        val scanFilter = filterServices
+            ?.map { ScanFilter.Builder().setServiceUuid(ParcelUuid(it)).build() }
+            ?.toList()
+        logger.info {
+            message = when (filterServices) {
+                null -> "Starting scan with no service filter."
+                else -> "Starting scan for services ${filterServices.joinToString()}."
+            }
+        }
+        scanner.startScan(scanFilter, scanSettings, callback)
 
         awaitClose {
-            bluetoothAdapter.bluetoothLeScanner.stopScan(callback)
+            logger.info {
+                message = when (filterServices) {
+                    null -> "Stopping scan with no service filter."
+                    else -> "Stopping scan for services ${filterServices.joinToString()}."
+                }
+            }
+            try {
+                scanner.stopScan(callback)
+            } catch (e: IllegalStateException) {
```

Comment:
Can we add a comment as to why we need to catch this exception here.
I seem to remember it being due to some Android weirdness, so it would be good to capture that and explain it here a little.

Reply:
tl;dr is that we can't check BLE state, only full Bluetooth state, so we need to try to stop the scan blindly. If it fails, it _probably_ just means that BLE is off.

Reply:
Why can't we check BLE state here?

Reply:
We can't check it anywhere. `BluetoothAdapter` internally keeps track of state, but filters it out before exposing it to apps.

From the Android sources:

```java
    /**
     * Get the current state of the local Bluetooth adapter.
     * <p>Possible return values are
     * {@link #STATE_OFF},
     * {@link #STATE_TURNING_ON},
     * {@link #STATE_ON},
     * {@link #STATE_TURNING_OFF}.
     *
     * @return current state of Bluetooth adapter
     */
    @RequiresPermission(Manifest.permission.BLUETOOTH)
    @AdapterState
    public int getState() {
        int state = getStateInternal();

        // Consider all internal states as OFF
        if (state == BluetoothAdapter.STATE_BLE_ON || state == BluetoothAdapter.STATE_BLE_TURNING_ON
                || state == BluetoothAdapter.STATE_BLE_TURNING_OFF) {
            if (VDBG) {
                Log.d(TAG, "Consider " + BluetoothAdapter.nameForState(state) + " state as OFF");
            }
            state = BluetoothAdapter.STATE_OFF;
        }
        if (VDBG) {
            Log.d(TAG, "" + hashCode() + ": getState(). Returning " + BluetoothAdapter.nameForState(
                    state));
        }
        return state;
    }
```